sdk/scheduler/src/main/java/org/apache/mesos/specification/DefaultContainerSpec.java

```diff
@@ -5,27 +5,35 @@
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 
-import javax.validation.constraints.NotNull;
+import javax.validation.Valid;
 import javax.validation.constraints.Size;
+import java.util.Optional;
 
 /**
  * Default implementation of {@link ContainerSpec}.
  */
 public class DefaultContainerSpec implements ContainerSpec {
-    @NotNull
     @Size(min = 1)
     private String imageName;
+    @Valid
+    private RLimitSpec rLimitSpec;
 
     @JsonCreator
     public DefaultContainerSpec(
-            @JsonProperty("image-name")
-            String imageName) {
+            @JsonProperty("image-name") String imageName,
+            @JsonProperty("rlimit_spec") RLimitSpec rLimitSpec) {
```

Comment:
You _might_ need to have these be optionals as well for de/reserialization to work as expected? It'd be good to have a unit test that checks de/reserialization works